File: FactoryPattern/LogisticsExample/Logistics/Creators/Logistics.cs
using Logistics.Interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Logistics.Creators
{
    public abstract class Logistics : ILogistics
    {
        public abstract ITransport CreateTransport();
        public void PlanDelivery() 
        {
        ITransport transport = CreateTransport();
            Console.WriteLine(transport.Deliver());
        }
    }
}

File: FactoryPattern/Java/src/creators/Logistics.java
package creators;

import interfaces.ILogistics;
import interfaces.ITransport;

public abstract class Logistics implements ILogistics {
    public abstract ITransport createTransport();

    @Override
    public void planDelivery() {
        ITransport transport = createTransport();
        System.out.println("Планиране на доставка...");
        System.out.println("Максимален товар: " + transport.getMaxCargoKg() + " кг");
        System.out.println(transport.deliver());
    }
}
